api/policy.go

```diff
@@ -143,22 +143,29 @@ func ParseUpdatePolicy(s string) (UpdatePolicy, error) {
 }
 
 type Policy struct {
-	PolicyID       string   `json:"policyId" yaml:"policyId"`
-	PolicyType     string   `json:"policyType" yaml:"-"`
-	QueryID        string   `json:"queryId" yaml:"queryId"`
-	Title          string   `json:"title" yaml:"title"`
-	Enabled        bool     `json:"enabled" yaml:"enabled"`
-	Description    string   `json:"description" yaml:"description"`
-	Remediation    string   `json:"remediation" yaml:"remediation"`
-	Severity       string   `json:"severity" yaml:"severity"`
-	Limit          int      `json:"limit" yaml:"limit"`
-	EvalFrequency  string   `json:"evalFrequency" yaml:"evalFrequency"`
-	AlertEnabled   bool     `json:"alertEnabled" yaml:"alertEnabled"`
-	AlertProfile   string   `json:"alertProfile" yaml:"alertProfile"`
-	Tags           []string `json:"tags" yaml:"tags"`
-	Owner          string   `json:"owner" yaml:"-"`
-	LastUpdateTime string   `json:"lastUpdateTime" yaml:"-"`
-	LastUpdateUser string   `json:"lastUpdateUser" yaml:"-"`
+	PolicyID               string                                               `json:"policyId" yaml:"policyId"`
+	PolicyType             string                                               `json:"policyType" yaml:"-"`
+	QueryID                string                                               `json:"queryId" yaml:"queryId"`
+	Title                  string                                               `json:"title" yaml:"title"`
+	Enabled                bool                                                 `json:"enabled" yaml:"enabled"`
+	Description            string                                               `json:"description" yaml:"description"`
+	Remediation            string                                               `json:"remediation" yaml:"remediation"`
+	Severity               string                                               `json:"severity" yaml:"severity"`
+	Limit                  int                                                  `json:"limit" yaml:"limit"`
+	EvalFrequency          string                                               `json:"evalFrequency" yaml:"evalFrequency"`
+	AlertEnabled           bool                                                 `json:"alertEnabled" yaml:"alertEnabled"`
+	AlertProfile           string                                               `json:"alertProfile" yaml:"alertProfile"`
+	Tags                   []string                                             `json:"tags" yaml:"tags"`
+	Owner                  string                                               `json:"owner" yaml:"-"`
+	LastUpdateTime         string                                               `json:"lastUpdateTime" yaml:"-"`
+	LastUpdateUser         string                                               `json:"lastUpdateUser" yaml:"-"`
+	ExceptionConfiguration map[string][]PolicyExceptionConfigurationConstraints `json:"exceptionConfiguration" yaml:"-"`
```

Comment:
I don't like this, but I couldn't find a better way to parse the Policy Exception constraints, as the format is: an Array of Maps in a Map. It's not clear to me why the outer map was required 🤷 
```
    "exceptionConfiguration": {
      "constraintFields": [
        {
          "dataType": "String",
          "fieldKey": "accountIds",
          "multiValue": true
        },
        {
          "dataType": "String",
          "fieldKey": "resourceNames",
          "multiValue": false
        },
        {
          "dataType": "KVTagPair",
          "fieldKey": "resourceTags",
          "multiValue": true
        }
      ]
    },
```

Reply:
Yeah, doesn't look necessary right now. Maybe they've plans to add another field to exceptionConfiguration.
Can Go be flexible like that - or does the structure/type need to be defined up front?

Reply:
Perhaps. Still annoying for parsing!
For parsing the values into structs we need to declare the structure up front. I was hoping I could tell GoLang to chain the map keys when parsing ie. `json:"exceptionConfiguration.constraintFields" yaml:"-"` but that doesn't seem to be possible.